bundler/lib/dependabot/bundler/native_helpers.rb

```diff
@@ -11,31 +11,28 @@ def self.run_bundler_subprocess(function:, args:, bundler_version:)
           function: function,
           args: args,
           # Kernel#spawn: clear environment variables except specified by env
-          unsetenv_others: true,
+          unsetenv_others: false,
           env: {
             # Bundler will pick the matching installed major version
             "BUNDLER_VERSION" => bundler_version,
-            # Required to find the ruby bin
-            "PATH" => ENV["PATH"],
-            # # Requried to create tmp directories in a writeable folder
-            "HOME" => ENV["HOME"],
-            # Required to git clone to a writeable folder
-            "GEM_HOME" => ENV["GEM_HOME"],
-            # Required by git fetch
-            "SSH_AUTH_SOCK" => ENV["SSH_AUTH_SOCK"],
-            # Env set by the runner
-            "SSL_CERT_FILE" => ENV["SSL_CERT_FILE"],
-            "http_proxy" => ENV["http_proxy"],
-            "HTTP_PROXY" => ENV["HTTP_PROXY"],
-            "https_proxy" => ENV["https_proxy"],
-            "HTTPS_PROXY" => ENV["HTTPS_PROXY"]
+            # Force bundler to use the helper gemfile that has been bundled with v1
+            "BUNDLE_GEMFILE" => File.join(versioned_helper_path(bundler_version: bundler_version), "Gemfile"),
+            # Unset ruby env set by running bundle exec from dependabot-core/bundler
+            "RUBYLIB" => nil,
+            "RUBYOPT" => nil,
+            "GEM_PATH" => nil,
+            "GEM_HOME" => nil
```

Comment:
@brrygrdn think this might work instead of unsetting the env, had to unset the the above, otherwise bundler got real confused about where it was loading gems and complained the helper gems hand't been installed when running `bundle exec`

Installed the gems with an explicit bundler version here: https://github.com/dependabot/dependabot-core/pull/3211/files#diff-a52f9842f7c3e55a30d022d6bc436a99d82b59aa42b14ccc61ca8aa8c3c2312bR27

This is slightly wasteful atm as they are only used for tests 😕